blockchain.py

```diff
@@ -148,33 +148,37 @@ def hash(block):
         block_string = json.dumps(block, sort_keys=True).encode()
         return hashlib.sha256(block_string).hexdigest()
 
-    def proof_of_work(self, last_proof):
+    def proof_of_work(self, last_block):
         """
         Simple Proof of Work Algorithm:
-         - Find a number p' such that hash(pp') contains leading 4 zeroes, where p is the previous p'
-         - p is the previous proof, and p' is the new proof
+         - Find a number p' such that hash(pp'h) contains leading 4 zeroes, where p is the previous p'
+         - p is the previous proof, p' is the new proof, and h is the hash of the last block
 
-        :param last_proof: <int>
+        :param last_block: <dict> last Block
         :return: <int>
         """
 
+        last_proof = last_block['proof']
+        last_hash = last_block['previous_hash']
```

Comment:
Don't you want the hash of last_block here, not last_block['previous_hash']? Otherwise, if we're mining the 2nd block, last_hash = 1 instead of the hash of the genesis block.

Reply:
Great point. I just push the change. Let me know if there's any other bugs.